src/imitation/util/networks.py

```diff
@@ -86,7 +86,13 @@ def forward(self, x: th.Tensor) -> th.Tensor:
             with th.no_grad():
                 self.update_stats(x)
 
-        return (x - self.running_mean) / th.sqrt(self.running_var + self.eps)
+        running_mean_ = (
+            self.running_mean[:, None, None] if len(x.shape) == 4 else self.running_mean
+        )
+        running_var_ = (
+            self.running_var[:, None, None] if len(x.shape) == 4 else self.running_var
+        )
+        return (x - running_mean_) / th.sqrt(running_var_ + self.eps)
```

Comment:
The return value of this flattened? Do we want that?

An alternative is to change `num_features` into `feature_shape` and just natively support non-flattened input vectors. Seems good to have general support!

That said, I'm not sure we even want to be normalizing image-based inputs in practice. They're already bounded after preprocessing between [0,1] -- does normalizing per-pixel even make sense given that?

Reply:
Per-channel normalization (which is what this does) seems like it makes sense to me. Re: flattening, that only happens when we use a NormalizeFeaturesExtractor, which flattens the input.

Regarding why to normalize at all, I think you're right that this doesn't matter in practice (altho you might be worried that something could get past `stable_baselines3.common.preprocessing.is_image_space` by having float elements or something), so probably makes sense to not normalize. Will remove the bits of code that handle normalization of image spaces.